modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/util/CommerceAccountRoleHelperImpl.java

```diff
@@ -220,6 +223,36 @@ else if (name.equals(AccountRoleConstants.ROLE_NAME_SUPPLIER)) {
 					new String[] {ActionKeys.ACCESS_IN_CONTROL_PANEL});
 			}
 
+			companyResourceActionIds.put(
+				"com.liferay.commerce.inventory",
+				new String[] {CommerceInventoryActionKeys.MANAGE_INVENTORY});
```

Comment:
Supplier should not have MANAGE_INVENTORY permission, it's too generic and it will cause other issues in the inventory section.
Please check what it is used for and find another solution

Reply:
Opted to use localService instead here: https://github.com/liferay-commerce/liferay-portal/commit/77f4ce8b5e1a4b9db76f9477f54b10a8817a5eb5